src/progress.rs

```diff
@@ -196,12 +196,15 @@ pub fn snapshot_progress(ui: &mut Ui) -> Option<impl Fn(&RepoPath) + '_> {
                     .output_guard(format!("\r{}", Clear(ClearType::CurrentLine))),
             );
         }
-        _ = write!(
-            state.ui,
-            "\r{}Snapshotting {}",
-            Clear(ClearType::CurrentLine),
-            path.to_fs_path(Path::new("")).display()
-        );
+
+        let bar_width = state.ui.term_width().map(usize::from).unwrap_or(20);
+        let msg: String = format!("Snapshotting {}", path.to_fs_path(Path::new("")).display())
+            .chars()
```

Comment:
Is `chars()` OK to use here, or should we use grapheme clusters?

Reply:
Rather than mucking with iterators at all, format with `"\r{}{:.*}", Clear(...), line_width, msg` (see [docs](https://doc.rust-lang.org/std/fmt/#precision))

Reply:
Oh, nice! TIL 😺

Reply:
OK, this looks much nicer (and now only has a single string format).